tests/std/tests/P0896R4_views_take_while/test.cpp

```diff
@@ -0,0 +1,493 @@
+    if constexpr (CanMemberEmpty<const R>) {
+        assert(as_const(r).empty() == is_empty);
+        assert(static_cast<bool>(as_const(r)) == !is_empty);
+    } else {
+        STATIC_ASSERT(CanEmpty<const R> == CanSize<const R>);
+        if constexpr (CanEmpty<const R>) {
+            assert(ranges::empty(as_const(r)) == is_empty);
+            assert(static_cast<bool>(as_const(r)) == !is_empty);
+        }
+    }
+
+    // Validate take_while_view::begin
+    STATIC_ASSERT(CanMemberBegin<R>);
+    STATIC_ASSERT(CanMemberBegin<const R>);
+    if (forward_range<V>) { // intentionally not if constexpr
+        const same_as<iterator_t<R>> auto i = r.begin();
+        if (!is_empty) {
+            assert(*i == *begin(expected));
+            assert(*r.begin() == *begin(expected));
+            assert(*r.begin() == *begin(expected));
+        }
+
+        if constexpr (copyable<V>) {
+            auto r2                              = r;
+            const same_as<iterator_t<R>> auto i2 = r2.begin();
+            if (!is_empty) {
+                assert(*i2 == *i);
+                assert(*r2.begin() == *i2);
+                assert(*r2.begin() == *i2);
+            }
+        }
+
+        const same_as<iterator_t<const R>> auto ic = as_const(r).begin();
+        if (!is_empty) {
+            assert(*ic == *begin(expected));
+            assert(*as_const(r).begin() == *begin(expected));
+            assert(*as_const(r).begin() == *begin(expected));
+        }
+
+        if constexpr (copyable<V>) {
+            auto rc2                                    = as_const(r);
+            const same_as<iterator_t<const R>> auto ic2 = rc2.begin();
+            if (!is_empty) {
+                assert(*ic2 == *ic);
+                assert(*rc2.begin() == *ic2);
+                assert(*rc2.begin() == *ic2);
+            }
+        }
+    }
+
+    // Validate take_while_view::end
+    STATIC_ASSERT(CanMemberEnd<R>);
+    STATIC_ASSERT(CanMemberEnd<const R>);
+    STATIC_ASSERT(CanEnd<const R&> == ranges::range<const V>);
+    STATIC_ASSERT(!common_range<R>);
+    STATIC_ASSERT(!common_range<const R>);
+    if (!is_empty) {
+        [[maybe_unused]] same_as<ranges::sentinel_t<R>> auto s        = r.end();
+        [[maybe_unused]] same_as<ranges::sentinel_t<const R>> auto sc = as_const(r).end();
+
+        if (forward_range<V>) { // intentionally not if constexpr
+            // Compare with const / non const iterators
+            assert(s != r.begin());
+            assert(s != as_const(r).begin());
+            assert(sc != r.begin());
+            assert(sc != as_const(r).begin());
+
+            // Compare with end of range
+            if constexpr (common_range<V>) {
+                assert(s == s.base());
+                assert(s == sc.base());
+                assert(sc == s.base());
+                assert(sc == sc.base());
+            } else {
+                if constexpr (forward_range<V>) {

```

Comment:
You could flatten this into `} else if constexpr (forward_range<V>) {`.

Reply:
No love for mountains?